container/factory/factory.go

```diff
@@ -16,7 +16,7 @@ import (
 
 // NewContainerManager creates the appropriate container.Manager for the
 // specified container type.
-func NewContainerManager(forType instance.ContainerType, conf container.ManagerConfig) (container.Manager, error) {
+var NewContainerManager = func(forType instance.ContainerType, conf container.ManagerConfig) (container.Manager, error) {
```

Comment:
Is this needed? I'm not sure why you would want a mutable `NewContainerManager`, unless you want to mock it out for tests? It's not obviously used else where in this PR.